FogCreek/Glitch-Community
# Functional politician: Replacing Default Collection Avatars

## Links
* Remix link: https://functional-politician.glitch.me/@scientiffic/lists
* Issue-tracker link: https://app.zenhub.com/workspaces/community-5d07d1359dee6059a9688176/issues/fogcreek/glitch-community-work/426
* Specs/doc links: https://docs.google.com/document/d/12elIoMtCI-si1rbzYs6khnNcxZvNWG79HlmR8Vol1A0/edit#

## GIF/Screenshots:
![avatar-generation](https://user-images.githubusercontent.com/519336/65465792-222f5c80-de2b-11e9-8790-52b5e8edfd99.gif)
Generated avatars appear on collection pages and are created based on the collection color, its complementary color, and the avatars of the first 1-3 projects in the collection.

If the complementary color has low contrast with the collection color, I pick either white or black as the contrast color:
![contrast](https://user-images.githubusercontent.com/519336/65466476-f1e8bd80-de2c-11e9-8e08-f989e6e5ab66.gif)

Avatars are removed elsewhere on collection cards:
![image](https://user-images.githubusercontent.com/519336/65465834-43904880-de2b-11e9-815a-d9d0779b60cf.png)
More by

![image](https://user-images.githubusercontent.com/519336/65465840-4a1ec000-de2b-11e9-8218-d0a873701928.png)
Profile Page

![image](https://user-images.githubusercontent.com/519336/65466262-53f4f300-de2c-11e9-97d7-13b46de020a7.png)
Search Results

For Search Autocomplete, a default avatar is used composed of the collection coverColor and the stacked border
![image](https://user-images.githubusercontent.com/519336/65466284-5bb49780-de2c-11e9-9137-99fd595ea278.png)

I also improved the info-hierarchy of collection cards on the homepage so that the name stick out more than the # of projects in the collection:
![image](https://user-images.githubusercontent.com/519336/65466523-19d82100-de2d-11e9-808b-4b78262615a6.png)


## How To Test:
Viewing collections - check that collection cards match the spec on these pages:
- user profile pages (https://functional-politician.glitch.me/@scientiffic/)
- project pages included in collections sections (https://functional-politician.glitch.me/~microbit-ble)
- collection pages (https://functional-politician.glitch.me/@scientiffic/microbit)
- search results (https://functional-politician.glitch.me/search?q=microbit&activeFilter=collection)

Editing Collections
- Sign into Functional Politician
- Go to one of your collection pages and try editing the color and see how the avatar changes
- Try adding / removing projects from a collection and check how the collection avatar changes

## Feedback I'm looking for:
- Did I break something?
- Are there ways I should be handling the collection SVG to ensure it's accessible?
- I haven't deleted adding the default avatar to new collection items because I'm scared it will break something.
